cdc/kv/client.go

```diff
@@ -1002,121 +998,146 @@ func (s *eventFeedSession) receiveFromStream(
 
 	// always create a new region worker, because `receiveFromStream` is ensured
 	// to call exactly once from outer code logic
-	worker := newRegionWorker(s.changefeed, s, addr)
-
-	defer worker.evictAllRegions()
+	worker := newRegionWorker(parentCtx, s.changefeed, s, addr)
+
+	ctx, cancel := context.WithCancel(parentCtx)
+	var retErr error
+	once := sync.Once{}
+	handleExit := func(err error) error {
```

Comment:
It seems unnecessary. Why not call `cancel` after `eg.Wait` directly?

Reply:
This is to be compatible with the previous error handling logic. Otherwise, many tests might have to be changed.